### ISSUE TITLE
Arabic question mark used with Adlam text falls back to a system font even when font contains the mark. 

### ISSUE THREAD

**Issue by NeilSureshPatel** [w3c/afrlreq#18]
A review of various browsers on desktop and mobile show the Arabic question mark rendered from a system font fall back despite the Adlam font containing the appropriate glyph. This can be seen at the center of the last line in the attached screen grabs. This was briefly discussed during [Meeting 7](https://github.com/w3c/font-text-cg/issues/39) W3C Font and Text Community Group. (Meeting notes in link) It is possibly related to a script itemization issue where Adlam is not properly recognized.

Chrome Mac OS:
<img width="712" alt="image" src="https://user-images.githubusercontent.com/46502494/106287333-ab6f5400-6214-11eb-8a52-0f25cac6e201.png">

Firefox Mac OS:
<img width="717" alt="image" src="https://user-images.githubusercontent.com/46502494/106287474-d5287b00-6214-11eb-9682-a9aa6b1f2bf1.png">

Edge Android:
![Screenshot_20210129-093413](https://user-images.githubusercontent.com/46502494/106288152-a8289800-6215-11eb-86ed-d9e92340c05c.png)

Firefox Android:
![Screenshot_20210129-093224](https://user-images.githubusercontent.com/46502494/106288166-ad85e280-6215-11eb-9693-0825b58063bd.png)

Chrome Android:
![Screenshot_20210129-093442](https://user-images.githubusercontent.com/46502494/106288177-b1196980-6215-11eb-8d19-519ba4731c37.png)

Sample Inspection results from Firefox showing the question mark is rendered with Times New Roman.
<img width="1436" alt="image" src="https://user-images.githubusercontent.com/46502494/106288580-3ac93700-6216-11eb-8cc0-c587762eaabb.png">

**Comment by andjc** [comment]
I wonder if this is inherited from UCD, looking at [Script Extensions](https://www.unicode.org/Public/UCD/latest/ucd/ScriptExtensions.txt), U+061F is associated with the following scripts: Arab, Rohg, Syrc, Thaa, Yezi. Have you tested in other African RTL scripts?

Although Arabic question mark fails in Yezidi text in Firefox, Works in Chrome and Edge, so I think there is a more fundamental issue with Firefox.

**Comment by NeilSureshPatel** [comment]
I have not tested other RTL African scripts. The two I can think of is Mende Kikakui, which I don't believe uses Arabic punctuation and Garay which has not been encoded yet.

**Comment by r12a** [comment]
N'Ko uses the Arabic question mark.  Experimenting with my character app, it seems afaict that the glyph is rendered as expected when using Conakry, but there is no QM glyph in the Noto font.

![Screenshot 2021-02-03 at 10 43 32](https://user-images.githubusercontent.com/4839211/106736210-1225b080-660d-11eb-9047-cc61a30e5e49.png)

![Screenshot 2021-02-03 at 10 45 48](https://user-images.githubusercontent.com/4839211/106736259-236ebd00-660d-11eb-8b2f-c6b1958b7c31.png)

**Comment by NeilSureshPatel** [comment]
Yes, I forgot about N'ko. I made a quick test on Windows since the default African system font is Ebrima which includes the QM. In these tests it looks like the incorrect font is used for the QM as well as the spaces. Spaces are rendered in the default body font while the QM is coming from a system font. This appears to be the case in Chrome as well. The glyph counts for each font matches Firefox.

Firefox:
![image](https://user-images.githubusercontent.com/46502494/106753773-1f856f80-65fa-11eb-8c68-05628f49e846.png)
![image](https://user-images.githubusercontent.com/46502494/106753883-4479e280-65fa-11eb-8370-e2bd71291654.png)
![image](https://user-images.githubusercontent.com/46502494/106753891-4774d300-65fa-11eb-896b-2c134112a8ff.png)

Chrome:
![image](https://user-images.githubusercontent.com/46502494/106753928-53609500-65fa-11eb-8f4f-18ff9fa44890.png)

**Comment by andjc** [comment]
Rendering engines more and more depend on UCD, so the absence of Adlam and N'ko for AQM in Script Extensions may be a factor.

**Comment by NeilSureshPatel** [comment]
Thanks. We discussed this a bit during the W3C Font and Text Community Group meeting and it was also confirmed that UCD Script Extensions plays a role in script itemization. I will move forward with getting the Script Extensions updated to include Adlam and N'ko to the appropriate Arabic punctuation marks.

**Comment by andjc** [comment]
Khaled in a recent tweet indicated that script itemisation was the responsibility of the client software, which should call the renderer indicating script and direction.

**Comment by NeilSureshPatel** [comment]
Another thing that Peter Constable brought up is that if a client has not been updated to even recognize a script that was recently encoded then it will by default treat the script as neutral. This can cause similar issues as well.

**Comment by r12a** [comment]
Does anyone have any news on this?  I see thata request was made to the Unicode Consortium to fix the [Script Extensions table](https://www.unicode.org/Public/15.0.0/ucd/ScriptExtensions.txt), and that appears to have been done.

I'm not getting any better results, though. See https://github.com/w3c/character_phrase_tests/issues/54

**Comment by NeilSureshPatel** [comment]
Hmm no news on this issue. I did a bit of testing, and I am seeing the same thing. The Script Extensions Table has a publish date in December of 2022. Is it possible that implementation can take a long time?

**Comment by andjc** [comment]
Script extensions have been updated but that doesn't necessarily mean that user agents have been updated

**Comment by r12a** [comment]
Ok.  I'll start raising browser bug reports then. Thanks.

**Comment by r12a** [comment]
Raised an issue for Gecko first, since @jfkthame may have insights into the required solution. If he does, i'll write more informed bug reports for Blink and WebKit.

There's a gap report at https://github.com/w3c/afrlreq/issues/19 to track this.
